### ISSUE TITLE
Failed push! adds empty bases

### ISSUE THREAD

**Issue by kescobo** [BioJulia/BioSequences.jl#115]
Is this intended?

```
julia> seq = dna"ATGC";

julia> push!(seq, "fail")
ERROR: MethodError: Cannot `convert` an object of type String to an object of type DNA
Closest candidates are:
  convert(::Type{DNA}, ::RNA) at /Users/ksb/.julia/packages/BioSymbols/O70zi/src/nucleicacid.jl:57
  convert(::Type{DNA}, ::Char) at /Users/ksb/.julia/packages/BioSymbols/O70zi/src/nucleicacid.jl:67
  convert(::Type{T}, ::UInt8) where T<:NucleicAcid at /Users/ksb/.julia/packages/BioSymbols/O70zi/src/nucleicacid.jl:50
  ...
Stacktrace:
 [1] enc64(::LongSequence{DNAAlphabet{4}}, ::String) at /Users/ksb/.julia/packages/BioSequences/k4j4J/src/longsequences/indexing.jl:125
 [2] unsafe_setindex! at /Users/ksb/.julia/packages/BioSequences/k4j4J/src/longsequences/indexing.jl:77 [inlined]
 [3] push!(::LongSequence{DNAAlphabet{4}}, ::String) at /Users/ksb/.julia/packages/BioSequences/k4j4J/src/biosequence/transformations.jl:23
 [4] top-level scope at REPL[27]:1
 [5] eval(::Module, ::Any) at ./boot.jl:331
 [6] eval_user_input(::Any, ::REPL.REPLBackend) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.4/REPL/src/REPL.jl:86
 [7] run_backend(::REPL.REPLBackend) at /Users/ksb/.julia/packages/Revise/tV8FE/src/Revise.jl:1165
 [8] top-level scope at none:0

julia> seq
5nt DNA Sequence:
ATGC-
```

If `push!()` throws an error, I would think that the sequence wouldn't change. This behavior seems like a footgun.

**Comment by kescobo** [comment]
FWIW, `append!()` works like I'd expect:

```
julia> append!(seq, "fail")
ERROR: MethodError: no method matching append!(::LongSequence{DNAAlphabet{4}}, ::String)
Closest candidates are:
  append!(::BitArray{1}, ::Any) at bitarray.jl:766
  append!(::AbstractArray{T,1} where T, ::Any) at array.jl:960
  append!(::DataStructures.MutableLinkedList, ::Any...) at /Users/ksb/.julia/packages/DataStructures/iWRBf/src/mutable_list.jl:161
  ...
Stacktrace:
 [1] top-level scope at REPL[30]:1
 [2] eval(::Module, ::Any) at ./boot.jl:331
 [3] eval_user_input(::Any, ::REPL.REPLBackend) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.4/REPL/src/REPL.jl:86
 [4] run_backend(::REPL.REPLBackend) at /Users/ksb/.julia/packages/Revise/tV8FE/src/Revise.jl:1165
 [5] top-level scope at none:0

julia> seq
5nt DNA Sequence:
ATGC-

julia> append!(seq, dna"a")
6nt DNA Sequence:
ATGC-A
```

**Comment by jakobnissen** [comment]
Should be fixed by #116 . In fact, `append!` does not work quite as well as you expect:

```
julia> s = dna"TAG"
3nt DNA Sequence:
TAG

julia> append!(s, aa"AA")
ERROR: MethodError: no method matching copyto!(::LongSequence{DNAAlphabet{4}}, ::Int64, ::LongSequence{AminoAcidAlphabet}, ::Int64, ::Int64)
[ etc etc ]

julia> s
5nt DNA Sequence:
TAG--
```
Unfortunately this is really difficult to fix, and is also an unfixable issue in base Julia (https://github.com/JuliaLang/julia/issues/15868)
